### PR TITLE
TE: refactoring anomaly feedback

### PR DESCRIPTION
**Pull request by puneetjaiswal**
normalizing anomaly_feedback table so that it can be used by othere entities as well

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/resources/migrations.xml [hunk 1]
```diff
@@ -334,4 +334,33 @@
...
+      </column>
+    </addColumn>
+
+    <dropColumn tableName="anomaly_results" columnName="feedback"/>
+
+    <createTable tableName="anomaly_feedback">
+      <column name="id" type="bigint" autoIncrement="true">
+        <constraints primaryKey="true" nullable="false"/>
+      </column>
+      <column name="feedback" type="varchar(100)"/>
```
Comment (jiateoh) [inline thirdeye/thirdeye-pinot/src/main/resources/migrations.xml]: rename to feedback_type so it's clearer that this is different from comments/description/status.

File: thirdeye/thirdeye-pinot/src/main/resources/migrations.xml [hunk 2]
```diff
@@ -334,4 +334,33 @@
...
+      <column name="comment" type="varchar(500)"/>
+      <column name="status" type="varchar(100)"/>
+    </createTable>
+    <sql dbms="mysql">
+      ALTER TABLE anomaly_feedback ENGINE = InnoDb;
+    </sql>
+    <addForeignKeyConstraint
+        constraintName="fk_anomaly_results_fk"
```
Comment (jiateoh) [inline thirdeye/thirdeye-pinot/src/main/resources/migrations.xml]: Use fk_anomaly_feedback_fk if the referenced table is feedback

Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/resources/migrations.xml]: renaming it to fk_anomaly_results_feedback_fk.

File: thirdeye/thirdeye-pinot/src/main/resources/migrations.xml (hunk 1 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/resources/migrations.xml]: sure.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/api/AnomalyFeedback.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    if (status != that.status) {
+      return false;
+    }
+    return comment != null ? comment.equals(that.comment) : that.comment == null;
+
+  }
+
+  @Override
+  public int hashCode() {
+    int result = (int) (id ^ (id >>> 32));
```
Comment (jiateoh) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/api/AnomalyFeedback.java]: use Objects.hash instead

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/api/AnomalyFeedback.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    AnomalyFeedback that = (AnomalyFeedback) o;
+
+    if (id != that.id) {
```
Comment (jiateoh) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/api/AnomalyFeedback.java]: Chain Objects.equals calls instead (see AnomalyFunctionSpec.java for an example).

**Comment by jiateoh** [comment]
LGTM - go ahead and push after addressing comments

**Pull request merged**